op-chain-ops/genesis/check.go

```diff
@@ -365,14 +376,94 @@ func PostCheckPredeployStorage(db vm.StateDB, finalSystemOwner common.Address, p
 }
 
 // PostCheckLegacyETH checks that the legacy eth migration was successful.
-// It currently only checks that the total supply was set to 0.
-func PostCheckLegacyETH(db vm.StateDB) error {
+// It checks that the total supply was set to 0, and randomly samples storage
```

Comment:
Curious how much time this saves in practice. Also wondering if the random sampling part should happen after the various checks that return early anyways

Reply:
Single-threaded iteration takes about 40 minutes. We can parallelize and read the whole state, but that will make the migration take longer.

I can do the state read as part of a separate PR - will require a refactor of the OVM_ETH migration itself to utilize a common lib for parallel state traversal.

Reply:
This already runs after the various checks that return early with the exception of the withdrawals migration.